src/structurizr/model/model.py

```diff
@@ -270,6 +271,21 @@ def add_container_instance(
         # TODO: implement
         # instance_number =
 
+    def add_component(
+        self, component: Optional[Component] = None, **kwargs,
+    ) -> Component:
+        name = component.name if component else kwargs["name"]
+        parent = kwargs["parent"]
+        if parent.get_component_with_name(name):
+            raise ValueError(f"{component} already exists for {parent}.")
+
+        if not component:
```

Comment:
I know this works but prefer comparison with `None` directly.

```suggestion
        if component is not None:
```